modules/tensor_mechanics/src/materials/ADMultiplePowerLawCreepStressUpdate.C

```diff
@@ -0,0 +1,144 @@
+  params.addRequiredParam<std::vector<Real>>("stress_thresholds",
+                                             "Stress intervals to switch creep behavior");
+  params.addRequiredParam<std::vector<Real>>("activation_energy", "Activation energy");
+  params.addParam<Real>("gas_constant", 8.3143, "Universal gas constant");
+  params.addParam<Real>("start_time", 0.0, "Start time (if not zero)");
+  return params;
+}
+
+ADMultiplePowerLawCreepStressUpdate::ADMultiplePowerLawCreepStressUpdate(
+    const InputParameters & parameters)
+  : ADRadialReturnCreepStressUpdateBase(parameters),
+    _temperature(isParamValid("temperature") ? &adCoupledValue("temperature") : nullptr),
+    _coefficient(getParam<std::vector<Real>>("coefficient")),
+    _n_exponent(getParam<std::vector<Real>>("n_exponent")),
+    _m_exponent(getParam<std::vector<Real>>("m_exponent")),
+    _stress_thresholds(getParam<std::vector<Real>>("stress_thresholds")),
+    _activation_energy(getParam<std::vector<Real>>("activation_energy")),
+    _gas_constant(getParam<Real>("gas_constant")),
+    _start_time(getParam<Real>("start_time")),
+    _exponential(1),
+    _exp_time(0),
+    _stress_index(0),
+    _number_of_models(_m_exponent.size())
+{
+
+  if (_coefficient.size() != _n_exponent.size() || _coefficient.size() != _m_exponent.size() ||
+      _coefficient.size() != _activation_energy.size())
+    paramError(
+        "n_exponent",
+        "Inputs to ADMultiplePowerLawCreepStressUpdate creep models must have the same size");
+
+  if (_number_of_models != _stress_thresholds.size() - 1)
+    paramError("stress_thresholds",
+               "The number of creep models must be the number of stress thresholds minute. That "
+               "is, one creep model will be valid between two stress thresholds");
+
+  for (unsigned int i = 0; i < _number_of_models; i++)
+    if (_start_time < this->_app.getStartTime() && (std::trunc(_m_exponent[i]) != _m_exponent[i]))
+      paramError("start_time",
+                 "Start time must be equal to or greater than the Executioner start_time if a "
+                 "non-integer m_exponent is used");
+}
+
+void
+ADMultiplePowerLawCreepStressUpdate::computeStressInitialize(
+    const ADReal & effective_trial_stress, const ADRankFourTensor & /*elasticity_tensor*/)
+{
+  _stress_index = stressIndex(effective_trial_stress);
+
+  if (_temperature)
+    _exponential =
+        std::exp(-_activation_energy[_stress_index] / (_gas_constant * (*_temperature)[_qp]));
+
+  _exp_time = std::pow(_t - _start_time, _m_exponent[_stress_index]);
+}
+
+ADReal
+ADMultiplePowerLawCreepStressUpdate::computeResidual(const ADReal & effective_trial_stress,
+                                                     const ADReal & scalar)
+{
+  if (_stress_index != stressIndex(effective_trial_stress))
+    mooseError("Internal issue in multiple creep model when selecting model according to stress "
+               "threshold");
```

Comment:
I don't quite get this. Will the effective trial stress not change between `computeStressInitialize` and successive calls to `computeResidual` ? If not then I'd rather make this an assert. Otherwise this should just be removed.

Reply:
Placing this error here was experimental. Interestingly, it's never hit for a simple problem or for an advanced practical case. 

I'll remove. We are just forcing it to stick to one creep branch during the entire return mapping logic, which I think it's fair.